src/warc2zim/url_rewriting.py

```diff
@@ -194,18 +192,15 @@ def normalize(url: HttpUrl) -> ZimPath:
 
     url_parts = urlsplit(url.value)
 
-    hostname = url_parts.hostname
+    if not url_parts.hostname:
+        raise Exception("Hostname is missing")
 
-    if hostname and hostname not in known_bad_hostnames:
-        try:
-            # try to decode the hostname
-            hostname = idna.decode(hostname)
-        except idna.IDNAError as exc:
-            # exception might happen if illegal character are found in hostname like the
-            # `_` in `host_ip` ; we keep the set of bad hostname in memory to not fill
-            # the logs with these warnings
-            logger.warning(f"Bad hostname found, kept as-is: {hostname}", exc_info=exc)
-            known_bad_hostnames.add(hostname)
+    # decode the hostname if it is punny-encoded
+    hostname = (
+        idna.decode(url_parts.hostname)
```

Comment:
With this we're not rewriting at all in case of IDNA parsing issue (on xn-- prefixed domain).
In such case, don't we want to still rewrite (reusing whatever bad domain there is)?

Reply:
As discussed live, the idea is that we do not trap any exception at this level because:
- if we are normalizing the URL from a WARC record, then hostname should be valid (browser achieved to fetch it) ; if it fails, then computing a badly formatted ZIM path is anyway error-prone and probably doomed to fail, so we let the error raise and it is catched upper in the stack and stops the scraper (and we will be able to revisit why this happened and fix this edge-case)
- if we are normalizing a URL found in a document, then when hostname is invalid then we know we will most probably not find a corresponding ZIM entry, so we let the error raise and it is catched upper in the stack and conducts the URL to be kept as-is